### PR TITLE
CMake parsing to process source nodes.

### PR DESCRIPTION
**Pull request by schmerl**

### CODE REVIEW

File: src/rosdiscover/project/models.py [hunk 1]
```diff
@@ -60,30 +61,37 @@ def close(self) -> None:
 
     def _recover(self, package: str, node: str) -> t.Optional[NodeModel]:
         # have we already recovered this model?
+        print(f"Recovering {package}/{node}")
```
Comment (ChrisTimperley) [inline src/rosdiscover/project/models.py:64]: prefer logger calls

File: src/rosdiscover/project/models.py [hunk 2]
```diff
@@ -60,30 +61,37 @@ def close(self) -> None:
...
-            package,
-            node,
-            node_info.entrypoint,
-            node_info.sources,
-            node_info.restrict_to_paths
-        )
+            try:
+                logger.info(f"Attempting to recover {package}/{node} from CMakeLists.txt")
+                model = self._recovery_tool.recover_using_cmakelists(package, node)
+            except ValueError as e:
```
Comment (ChrisTimperley) [inline src/rosdiscover/project/models.py:73]: We may also want to catch `RuntimeError` here, too. (We use that in cases where the input is fine but the C++ static recovery failed.)

File: src/rosdiscover/project/models.py [hunk 3]
```diff
@@ -60,30 +61,37 @@ def close(self) -> None:
...
+                logger.error(f"Error recovering {package}/{node}: {e}")
```
Comment (ChrisTimperley) [inline src/rosdiscover/project/models.py:74]: We can use `logger.exception` to record the stack trace here. (We can then drop `{e}` from the message.)

File: src/rosdiscover/project/models.py (hunk 1 above)
Comment (schmerl) [reply src/rosdiscover/project/models.py:64]: Done

File: src/rosdiscover/project/models.py (hunk 2 above)
Comment (schmerl) [reply src/rosdiscover/project/models.py:73]: Done

File: src/rosdiscover/project/models.py (hunk 3 above)
Comment (schmerl) [reply src/rosdiscover/project/models.py:74]: Done.

**Pull request merged**